internal/venice-common/src/main/java/com/linkedin/venice/pubsub/manager/TopicManager.java

```diff
@@ -651,72 +618,110 @@ public synchronized boolean containsTopicAndAllPartitionsAreOnline(
+   * @return a Map of partition to the latest offset, or an empty map if there's any problem getting the offsets
+   */
+  public Int2LongMap getTopicLatestOffsets(PubSubTopic pubSubTopic) {
+    return topicMetadataFetcher.getTopicLatestOffsets(pubSubTopic);
+  }
+
+  /**
+   * Get partition count for a given topic.
+   * @param pubSubTopic  the topic to get partition count for
+   * @return the number of partitions for the given topic
+   * @throws PubSubTopicDoesNotExistException if the topic does not exist
    */
-  public Int2LongMap getTopicLatestOffsets(PubSubTopic topic) {
-    return partitionOffsetFetcher.getTopicLatestOffsets(topic);
+  public int getPartitionCount(PubSubTopic pubSubTopic) {
+    List<PubSubTopicPartitionInfo> partitionInfoList = getTopicPartitionInfo(pubSubTopic);
+    if (partitionInfoList == null) {
+      throw new PubSubTopicDoesNotExistException("Topic: " + pubSubTopic + " does not exist");
+    }
+    return partitionInfoList.size();
+  }
+
+  public boolean containsTopic(PubSubTopic pubSubTopic) {
+    return topicMetadataFetcher.containsTopic(pubSubTopic);
+  }
+
+  public boolean containsTopicCached(PubSubTopic pubSubTopic) {
+    return topicMetadataFetcher.containsTopicCached(pubSubTopic);
   }
 
-  public long getPartitionLatestOffsetAndRetry(PubSubTopicPartition pubSubTopicPartition, int retries) {
-    return partitionOffsetFetcher.getPartitionLatestOffsetAndRetry(pubSubTopicPartition, retries);
+  public long getLatestOffsetWithRetries(PubSubTopicPartition pubSubTopicPartition, int retries) {
+    return topicMetadataFetcher.getLatestOffsetWithRetries(pubSubTopicPartition, retries);
   }
 
-  public long getProducerTimestampOfLastDataRecord(PubSubTopicPartition pubSubTopicPartition, int retries) {
-    return partitionOffsetFetcher.getProducerTimestampOfLastDataRecord(pubSubTopicPartition, retries);
+  public long getLatestOffsetCached(PubSubTopic pubSubTopic, int partitionId) {
+    return topicMetadataFetcher.getLatestOffsetCached(new PubSubTopicPartitionImpl(pubSubTopic, partitionId));
+  }
+
+  public long getProducerTimestampOfLastDataMessageWithRetries(PubSubTopicPartition pubSubTopicPartition, int retries) {
+    return topicMetadataFetcher.getProducerTimestampOfLastDataMessageWithRetries(pubSubTopicPartition, retries);
+  }
+
+  public long getProducerTimestampOfLastDataMessageCached(PubSubTopicPartition pubSubTopicPartition) {
+    return topicMetadataFetcher.getProducerTimestampOfLastDataMessageCached(pubSubTopicPartition);
   }
 
   /**
    * Get offsets for only one partition with a specific timestamp.
    */
-  public long getPartitionOffsetByTime(PubSubTopicPartition pubSubTopicPartition, long timestamp) {
-    return partitionOffsetFetcher.getPartitionOffsetByTime(pubSubTopicPartition, timestamp);
+  public long getOffsetByTime(PubSubTopicPartition pubSubTopicPartition, long timestamp) {
+    return topicMetadataFetcher.getOffsetForTimeWithRetries(pubSubTopicPartition, timestamp, 25);
```

Comment:
how do we come up with `25` here?

Reply:
It is from the current/existing code

Reply:
I see.. `25` looks weird, but it is fine to leave it this way since it exists before the PR.